lib/msf/core/modules/loader/executable.rb

```diff
@@ -11,13 +11,38 @@ def loadable?(path)
     File.directory?(path)
   end
 
-  def loadable_module?(parent_path, type, module_reference_name)
-    full_path = module_path(parent_path, type, module_reference_name)
+  # @param [String] parent_path Root directory to load modules from
+  # @param [String] type Such as auxiliary, exploit, etc
+  # @param [String] module_reference_name The module reference name, without the type prefix
+  # @param [nil,Msf::Modules::Metadata::Obj] cached_metadata
+  # @return [Boolean] True this loader can load the module, false otherwise
+  def loadable_module?(parent_path, type, module_reference_name, cached_metadata: nil)
+    full_path = cached_metadata&.path || module_path(parent_path, type, module_reference_name)
     script_path?(full_path)
   end
 
   protected
 
+  def read_script_runtime(full_path)
+    # Extract the runtime from the first line of the script, i.e.
```

Comment:
```suggestion
    # Extract the runtime from the shebang, i.e.
```

https://en.wikipedia.org/wiki/Shebang_(Unix)

Reply:
I went with this terminology for Go's semantics: https://unix.stackexchange.com/questions/162531/shebang-starting-with :+1:

Reply:
TIL